Photo Selector 2

OmarBasem opened the pull request:
fixes #14474

This is a second PR for the implementation of the new photo selector. The PR implements reading more images from the device as the user scrolls down the selector.

Designs: https://www.figma.com/file/wA8Epdki2OWa8Vr067PCNQ/Composer-for-Mobile?node-id=2827%3A269831&t=1NrhQoSGitjVl9dD-4

OmarBasem committed:
updates

OmarBasem committed:
lint

OmarBasem committed:
refactor

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | a9acd911 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14487/1/) | 2022-12-05 06:54:59 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14487/1/consoleText) |

OmarBasem committed:
lint

ibrkhalil commented on Makefile:
```diff
@@ -319,7 +319,7 @@ run-visual-test-ios: XCODE_DERIVED_DATA := $(HOME)/Library/Developer/Xcode/Deriv
 run-visual-test-ios: APPLICATION_NAME := StatusIm-brfnruzfrkkycpbndmdoeyrigthc
 run-visual-test-ios: export TEST_BINARY_PATH := $(XCODE_DERIVED_DATA)/$(APPLICATION_NAME)/Build/Products/Debug-iphonesimulator/StatusIm.app
 run-visual-test-ios: ##@test Run tests once in NodeJS
-	detox test --configuration ios.sim.debug 
```
Is this related?

ibrkhalil commented on src/status_im/ui2/screens/chat/composer/view.cljs:
```diff
@@ -138,13 +162,15 @@
                   added-value          (if (and (not (seq suggestions)) (or edit reply)) 38 0) ; increased height of input box needed when reply
                   min-y                (+ min-y (when (or edit reply) 38))
                   y                    (get-y-value context keyboard-shown min-y max-y added-value max-height chat-id suggestions reply)
+                  y                    (+ y (when (seq images) 80))
```
duplicated var ?

ibrkhalil commented on src/status_im/ui2/screens/chat/photo_selector/view.cljs:
```diff
@@ -78,14 +55,21 @@
 
+(defn on-end-reached [end-cursor]
+  (let [is-loading    (rf/sub [:camera-roll-loading-more])
+        has-next-page (rf/sub [:camera-roll-has-next-page])]
```
bools?
Should we add a question mark at the end ?

flexsurfer commented on src/status_im/ui2/screens/chat/composer/style.cljs:
```diff
@@ -80,3 +80,14 @@
                          :left     34
                          :top      3})))
+
+(defn remove-photo-container []
```
def

flexsurfer commented on src/status_im/ui2/screens/chat/photo_selector/view.cljs:
```diff
@@ -35,24 +23,13 @@
-                                      (reset! selected [])
-                                      (rf/dispatch [:bottom-sheet/hide]))}
-            (str (i18n/label :t/send) " " (when (> (count @selected) 1) (count @selected)))]]])))])
+          [quo2/button {:style    {:width "95%"
```
we shouldn't use percentage, only margins and paddings

flexsurfer commented on src/status_im/ui2/screens/chat/photo_selector/view.cljs:
```diff
@@ -35,24 +23,13 @@
+          [quo2/button {:style    {:width "95%"
+                                   :align-self :center}
+                        :on-press #(do
+                                     ;(rf/dispatch [:chat.ui/send-current-message])
```
comment?

flexsurfer commented on src/status_im/ui2/screens/chat/composer/view.cljs:
```diff
@@ -117,6 +120,27 @@
                  translate-y
                  (reanimated/with-timing (- max-y)))))))))))
 
+(defn small-image [item]
```
better to move  to separate namespace

flexsurfer commented on src/status_im/ui2/screens/chat/composer/view.cljs:
```diff
@@ -117,6 +120,27 @@
+(defn small-image [item]
+  [rn/view
+   [rn/image {:source {:uri (first item)}
+              :style  {:width         56
```
move to style ns

ilmotta commented on src/status_im/chat/models/images.cljs:
```diff
@@ -119,13 +121,23 @@
+  [{:keys [db] :as cofx} photos page-info end-cursor]
+  (let [photos_x (when end-cursor (:camera-roll-photos db))]
+    {:db (-> db
+             (assoc :camera-roll-photos (concat photos_x (mapv #(get-in % [:node :image :uri]) photos)))
```
Using `mapv` followed by `concat` is unnecessary, because `concat` will convert the vector back to a lazy seq. Just `map` suffices.

```clojure
user=> (concat [1 2 3] [4 5 6])
(1 2 3 4 5 6)
user=> (concat '(1 2 3) '(4 5 6))
(1 2 3 4 5 6)
```

OmarBasem replied:
I need to concat the photos that were read previously with the newly read photos

ilmotta replied:
But what I mean is that it's pointless to call `mapv` (see the `v` in the end). You don't need to convert to a vector if you're going to call `concat`, just use `map` (without the `v`). See my previous example.

ilmotta commented on src/status_im/chat/models/images.cljs:
```diff
@@ -119,13 +121,23 @@
+  [{:keys [db]} is-loading]
+  {:db (assoc db :camera-roll-loading-more is-loading)})
 
 (fx/defn on-camera-roll-get-photos
```
We have ways of testing event handlers and there are examples in the codebase, I would say we should almost always test them. One of the principles of the https://deviq.com/principles/boy-scout-rule is to try to at least not increase tech debt, and if possible reduce it.

ilmotta commented on src/status_im/chat/models/images.cljs:
```diff
@@ -100,13 +100,15 @@
-                    (-> (.getPhotos CameraRoll #js {:first num :assetType "Photos" :groupTypes "All"})
-                        (.then #(re-frame/dispatch [:on-camera-roll-get-photos (:edges (types/js->clj %))]))
-                        (.catch #(log/warn "could not get cameraroll photos"))))})))
+                    (-> (if end-cursor
```
I usually prefer to see effects that abstract the `on-success` and `on-failure` handlers, so they become more general and are easier to test. This is a known pattern, you can see it, for instance in the http-fx effect https://github.com/day8/re-frame-http-fx/blob/master/src/day8/re_frame/http_fx.cljs#L26

But the really important feedback I have for this effect is that it lacks testing, and since you're touching it, it'd be great to have them.

OmarBasem replied:
I could be wrong, but I do not think we can use `on-success` and `on-failure` here. This is a JS promise.

As for testing, `permissions/request-permissions` requires reading from an actual device/simulator.

ilmotta replied:
@OmarBasem, sorry this is a relatively long topic. Please, don't worry about re-implementing this effect in this PR, it seems to be out of scope for now.

Nevertheless, this topic is quite important, so I'm going to expand a little bit here.

What I mean by my previous comment is that this effect could abstract what to do on the success and fail conditions. This is the recommended approach for effects, i.e. the effect would still use a promise, no problem with that because that's why it exists, to abstract effectful handlers, but consumers of the effect would be able to hook into the success/fail conditions using just raw data. This is usually desirable for effects, because we can make them more flexible and easier to test.

It's important to repeat that re-frame strongly recommends that **all effects should be represented as data**, not data + anonymous functions. This is a mistake in the current implementation of effects in our codebase and it makes testing and inspecting effects harder.

> As for testing, permissions/request-permissions requires reading from an actual device/simulator.

We can always stub hard OS dependencies.

So the following is just an idea I coded now to help you understand what a better effect should look like.  Btw, I haven't tested the implementation below, and there's an edge case with `permissions/request-permissions`, because it incorrectly *sometimes returns a promise, sometimes it doesn't*.

For example:

```clojure
(re-frame/reg-fx
 :camera-roll/get-photos
 (fn [{:keys [num end-cursor on-success on-failure]}]
   (-> (permissions/request-permissions
        {:permissions [:read-external-storage]
         :on-allowed  (fn []
                        (-> (if end-cursor
                              (.getPhotos CameraRoll #js {:first      num
                                                          :after      end-cursor
                                                          :assetType  "Photos"
                                                          :groupTypes "All"})
                              (.getPhotos CameraRoll #js {:first      num
                                                          :assetType  "Photos"
                                                          :groupTypes "All"}))))})
       (.then (fn [^js js-result]
                (when on-success
                  (let [result (types/js->clj js-result)]
                    (re-frame/dispatch (conj on-success {:edges     (:edges result)
                                                         :page-info (:page_info result)}))))))
       (.catch (fn [error]
                 (when on-failure
                   (re-frame/dispatch (conj on-failure error)))) ))))
```

And here's how consumers would more or less use this effect in event handlers. Notice that `on-success` and `on-failure` are **vectors, not anonymous functions**. Everything is data ;)

```clojure
{:camera-roll/get-photos {:num num
                          :end-cursor end-cursor
                          :on-success [:on-camera-roll-get-photos end-cursor]
                          :on-failure [:camera-roll/get-photos-failure]}}
```

---

An evolution of this effect could/should get permissions using an interceptor. Effectful inputs as data as well ;)

OmarBasem replied:
@ilmotta Thank you so much for this detailed explanation! ❤️ Lots of useful information.

I have one question, so here we have `.then` followed by `on-success`, like is there any other option? Can't we just drop out the `on-success` and `on-failure` since we are using `then` and `catch`?

ilmotta replied:
@OmarBasem, we can definitely drop the `on-success` and `on-failure` options. It's up to the developer to choose how reusable the effect is. If you want to design an effect that other events can dispatch and react accordingly, then having `on-success`/`on-failure` is the recommended approach in re-frame. But if you just want other events to *fire and forget* (dispatch `::camera-roll/get-photos`) then you can drop those options.

Both alternatives are valid. Without `on-success`/`on-failure` it's also a bit harder to test, but still doable.

ilmotta commented on src/status_im/chat/models/images.cljs:
```diff
@@ -77,17 +77,17 @@
     (fn [^js images]
-      ;; NOTE(Ferossgp): Because we can't highlight the already selected images inside
-      ;; gallery, we just clean previous state and set all newly picked images
+        ;; NOTE(Ferossgp): Because we can't highlight the already selected images inside
```
There seems to be extra whitespace here.

ilmotta commented on src/status_im/ui2/screens/chat/composer/view.cljs:
```diff
@@ -117,6 +120,27 @@
+    [quo2/icon :i/close {:color colors/white :size 12}]]])
+
+(defn images-list [images]
+  [rn/flat-list {:key-fn     (fn [item] (first item))
```
Just `first` suffices as the `:key-fn`.

du64 commented:
Would you please make the figma files public so I can have a reference to look at? thanks

OmarBasem commented:
> Would you please make the figma files public so I can have a reference to look at? thanks

I do not have the privilege to do so. I will ask the design team.

OmarBasem committed:
tests

OmarBasem committed:
refactor

OmarBasem committed:
lint

du64 commented:
The selected photos hide when opening a chat

https://user-images.githubusercontent.com/65834069/206220629-9d9758bc-0dd7-47c8-9c3f-93a8a68d4a65.mp4

du64 commented:
Must swipe up to view the confirmation button. Tapping it requires releasing the bottom sheet while tapping repeatedly in the general area in hopes that it will recognize the button tap.

https://user-images.githubusercontent.com/65834069/206365613-a50571c8-ca1c-4c76-af27-a65589f782b3.mp4

ilmotta commented on src/status_im/chat/models/images.cljs:
```diff
@@ -100,13 +100,15 @@
+                    (-> (if end-cursor
+                          (.getPhotos CameraRoll #js {:first num :after end-cursor :assetType "Photos" :groupTypes "All"})
+                          (.getPhotos CameraRoll #js {:first num :assetType "Photos" :groupTypes "All"}))
+                        (.then #(re-frame/dispatch [:on-camera-roll-get-photos (:edges (types/js->clj %)) (:page_info (types/js->clj %)) end-cursor]))
```
You could extract the expensive (?) data converstion of `types/js->clj` into a binding.

ilmotta commented on src/status_im/chat/models_test.cljs:
```diff
@@ -91,3 +92,8 @@
       (let [res (chat/navigate-to-chat-nav2 {:db db} chat-id false)]
         (is (nil? (get-in res [:db :pagination-info chat-id :all-loaded?])))))))
+
+(deftest camera-roll-loading-more
```
Just a missing `-test` suffix. `deftest` creates a var, so it's conventional to add a suffix.

ilmotta commented on src/status_im2/subs/root.cljs:
```diff
@@ -100,6 +100,9 @@
 (reg-root-key-sub :chat/inputs :chat/inputs)
 (reg-root-key-sub :chat/memberships :chat/memberships)
 (reg-root-key-sub :camera-roll-photos :camera-roll-photos)
+(reg-root-key-sub :camera-roll-end-cursor :camera-roll-end-cursor)
```
Not suggesting anything here, but does anybody know if adding more root keys to the graph can meaningfully impact performance?

Is it generally better to have less root keys? Would it better for the performance of the app to create just one `:camera-roll` root key and have the `next-page` and `loading-more` keys inside it?

My intuition tells me that it's better to have less root keys for things that are interconnected, so that we can better leverage subscription caching.

@flexsurfer cc

flexsurfer replied:
yes i just had the same feeling lately reviewing PRs,  we should group props in root keymap

flexsurfer replied:
would be cool to add this to guidelines

ilmotta commented on src/status_im/ui2/screens/chat/photo_selector/view.cljs:
```diff
@@ -78,14 +54,21 @@
    (when (some #{item} @selected)
      [info-count/info-count (+ (utils/first-index #(= item %) @selected) 1) (style/image-count)])])
 
+(defn on-end-reached [end-cursor]
```
@OmarBasem, I really think this kind of logic should be in event handlers. So you can define an event, and in the `:on-end-reached` of the flat list you can dispatch it. For an example, see `src/status_im/ui/screens/activity_center/views.cljs`, function `activity center`.

OmarBasem replied:
I think that's what I am doing. The difference is that here we are dispatching based on a condition first (?)

ilmotta replied:
What I mean is that this function shouldn't be part of UI namespaces, does it make sense? It seems to me the function should be defined as an event handler instead.

OmarBasem replied:
@ilmotta I think it is better to keep it that way where it checks for conditions before dispatching. Otherwise, it would be unnecessarily dispatching events (in a spamful way) every time it reaches `end`

Also, the more events are dispatched, the more the performance may be impacted.

ilmotta replied:
I disagree @OmarBasem, I think this is real misconception in the code.

1. Subscriptions have a cost and they should ideally be left for UI components according to re-frame. `on-end-reached` is not a UI component. In fact, it's just a function that needs to read some state and dispatch two effects. This is exactly what an event handler is supposed to do.

2. The `on-end-reached` function will need to be called by re-frame every time either `camera-roll/loading-more` or `camera-roll/has-next-page` state changes and then further dispatch. Every function that subscribes to something will need to be potentially called multiple times due to rendering cycles. This is more work than only dispatching if the RN FlatList detects the onEndReachedThreshold has been reached.

3. If we follow this very logic we would then have to move lots and lots of event handlers to the view layer and rely on needless subscriptions. This would dramatically hurt the performance of the app and break re-frame's recommendations.

OmarBasem replied:
Thanks @ilmotta for the explanation. Fixed it.

ilmotta commented on shadow-cljs.edn:
```diff
@@ -82,6 +82,7 @@
            :output-dir       "target/test"
            :optimizations    :simple
            :target           :node-test
+           :ns-regexp        "status-im.chat.models-test$"
```
This is probably a left over from your local experiments.

ilmotta replied:
May I suggest you comment this out and add I quick explanation? This is quite handy for people running tests without the REPL when using `make test-watch`, so I think we could leave it here (commented) for other devs.

OmarBasem committed:
refactor

OmarBasem committed:
refactor

OmarBasem committed:
refactor

OmarBasem committed:
refactor

OmarBasem committed:
updates

OmarBasem committed:
lint

OmarBasem committed:
lint

flexsurfer commented on src/status_im/ui2/screens/chat/composer/composer_images/composer_images.cljs:
```diff
@@ -0,0 +1,24 @@
+(ns status-im.ui2.screens.chat.composer.composer-images.composer-images
```
no need to duplicate `composer` in the name, its already under `composer` namespace, and also it should be view.cljs so something like `status-im.ui2.screens.chat.composer.images.view`

flexsurfer replied:
btw you could create it in status-im2 , so we don't need to move it

OmarBasem replied:
Better move the entire composer together?

flexsurfer commented on src/status_im/ui2/screens/chat/composer/composer_images/composer_images.cljs:
```diff
@@ -0,0 +1,24 @@
+            [re-frame.core :as rf]))
+
+(defn image [item]
+  [rn/view
```
`[:<>` better to avoid using empty views

OmarBasem replied:
It's not an empty view really. It does not look right if I use `[:<>` instead of `view`. There is a button that is positioned absolutely relative to that view.

flexsurfer commented on src/status_im2/subs/root.cljs:
```diff
@@ -99,7 +99,10 @@
 (reg-root-key-sub :chat/inputs :chat/inputs)
 (reg-root-key-sub :chat/memberships :chat/memberships)
-(reg-root-key-sub :camera-roll-photos :camera-roll-photos)
+(reg-root-key-sub :camera-roll/photos :camera-roll/photos)
```
its better to have `:camera-roll` map instead of a bunch of root keys

J-Son89 commented on shadow-cljs.edn:
```diff
@@ -82,6 +82,8 @@
            :output-dir       "target/test"
            :optimizations    :simple
            :target           :node-test
+           ;; When running tests without a REPL you can uncomment the below line to `make test-watch` a specific file
```
can this just be passed in as a parameter to the `make test-watch` command ??

ilmotta replied:
Yes, one way is via a environment variable, because ShadowCLJS supports this in the `edn` file.

So for example, something like this should be doable:

```bash
NS_REGEX='status-im.chat.models-test$' make test-watch
```

ilmotta replied:
@J-Son89 ^

pavloburykh commented:
@OmarBasem Hi! Could you please resolve existing conflicts and PR to the latest develop. After that we can take it into test.  Thanx!

OmarBasem committed:
refactor

OmarBasem committed:
rebase

OmarBasem committed:
updates

du64 commented:
> Must swipe up to view the confirmation button. Tapping it requires releasing the bottom sheet while tapping repeatedly in the general area in hopes that it will recognize the button tap.
> 
>  22-12-07-23-38-49.mp4

Still having this issue

OmarBasem commented:
> > Must swipe up to view the confirmation button. Tapping it requires releasing the bottom sheet while tapping repeatedly in the general area in hopes that it will recognize the button tap.
> > 22-12-07-23-38-49.mp4
> 
> Still having this issue

That's an issue with the bottom sheet component. Not part of this PR.

OmarBasem commented:
@pavloburykh ready to be tested

OmarBasem committed:
lint

status-im-auto commented:
## 88% of end-end tests have passed
```
Total executed tests: 17
Failed tests: 2
Passed tests: 15
Not executed tests: 3
```
```
IDs of not executed tests: 702807,702808,702847 
```
```
IDs of failed tests: 702840,702844 
```
<h3>Not executed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14487.apk&PR_ID=14487&TR_CASE_IDS=702807,702808,702847">Rerun not executed tests</a></li></details><h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14487.apk&PR_ID=14487&TR_CASE_IDS=702840,702844">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068155">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `RepliedMessageText` by `xpath`: `//*[starts-with(@text,'test message')]/preceding::android.widget.TextView[@content-desc='quoted-message']` </div><div>Device 1: `RepliedMessageText` is `👴🏼` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:693: in test_community_emoji_send_copy_paste_reply
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Emoji message was not copied
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#183">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61/0073screenshot.png?auth=6e286e13f2cab8954337a29abe02d855">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#112">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea/0051screenshot.png?auth=f2be2f4d5b24a0d5f5da56656368d25a">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068159">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><p><blockquote><div>Device 2: Getting preview message for link: https://giphy.com/gifs/this-is-fine-QMHoU66sBXqqLqYvGO </div><div>Device 2: Scrolling down to PreviewImage </div></blockquote></p>

```
critical/test_public_chat_browsing.py:758: in test_community_links_with_previews_github_youtube_twitter_gif_send_enable
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Enable button is still shown after clicking on 'Don't ask again'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#240">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61/0073screenshot.png?auth=6e286e13f2cab8954337a29abe02d855">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#133">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea/0051screenshot.png?auth=f2be2f4d5b24a0d5f5da56656368d25a">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (15)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068160">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#287">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#193">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068156">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#210">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068154">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#161">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068157">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#267">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#185">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068153">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068158">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/215ca800d77a44c08e2ac013e6369d61?auth=6e286e13f2cab8954337a29abe02d855#146">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83ccfbae3dbe49e2a7e85f48ac0976ea?auth=f2be2f4d5b24a0d5f5da56656368d25a#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068161">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/64f3c02277484012a639780e89c91e07?auth=39909a0acbf71f6d6300aa8363ee55da#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068145">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d4853ede2fbf460686a8eb57cdd8ac85?auth=4b659ef2f019acbbc153959f91312728#65">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/50a7f4f0d92f4a78bfe118f79be85628?auth=2444c15f2c5f5205331f92bab7280991#60">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068150">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d8adc836849a42d0bbdbfe10f56da333?auth=9d3e20f7900a741e1fd63ca977d06525#80">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068148">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d8adc836849a42d0bbdbfe10f56da333?auth=9d3e20f7900a741e1fd63ca977d06525#41">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068149">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d8adc836849a42d0bbdbfe10f56da333?auth=9d3e20f7900a741e1fd63ca977d06525#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068146">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5aed4d3b320e4335adb53fc5341ce4e1?auth=efcfc5496a3dac290a7d0c10681810f3#271">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6daf49751b5b4982b10456cd57da644a?auth=ec5c966675c55f595033d0d9eba207c5#174">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068147">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5aed4d3b320e4335adb53fc5341ce4e1?auth=efcfc5496a3dac290a7d0c10681810f3#192">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6daf49751b5b4982b10456cd57da644a?auth=ec5c966675c55f595033d0d9eba207c5#109">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068143">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5aed4d3b320e4335adb53fc5341ce4e1?auth=efcfc5496a3dac290a7d0c10681810f3#84">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6daf49751b5b4982b10456cd57da644a?auth=ec5c966675c55f595033d0d9eba207c5#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068144">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5aed4d3b320e4335adb53fc5341ce4e1?auth=efcfc5496a3dac290a7d0c10681810f3#104">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6daf49751b5b4982b10456cd57da644a?auth=ec5c966675c55f595033d0d9eba207c5#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

VladimrLitvinenko commented:
Hi @OmarBasem thank you for your work. Here are the following found issues:
 
**ISSUE 1 The 'confirm selection' button is not shown for android devices**
**Steps to reproduce:**
1. Open 1-1 chat
2. tap image icon
3. Select any photo within device

**Actual result:**
The 'confirm selection' button is not shown
![image](https://user-images.githubusercontent.com/52490791/208085499-e26927a0-05f0-499a-971e-a571130b68d0.png)

**Expected result:**
The 'confirm selection' button is shown
![image](https://user-images.githubusercontent.com/52490791/208085739-4b9d1d03-3387-49db-bc25-ab2a10380587.png)

**Env: Samsung Galaxy S20 v10, Huawei p20 light v9**


**ENV:** Samsung Galaxy S20 v10

**ISSUE 2 The header is visible during selector is opened**
**Steps to reproduce:**
1. Open 1-1 chat
2. tap image icon

**Actual result:**
Header is shown when selector is opened
![image](https://user-images.githubusercontent.com/52490791/208088563-2cef7bed-b1ca-4e7a-81f5-ab69083b1b52.png)


**Expected result:**
Header is not shown per design
![image](https://user-images.githubusercontent.com/52490791/208089592-7bfd555f-ba52-4c14-8013-18a81f1ec9b9.png)

**Env: Samsung Galaxy S20 v10, Huawei p20 light v9**

**ISSUE 3: Images are sent in the message even if [confirm selection] button is not tapped**
**Steps to reproduce:**
1. Open 1-1 chat
2. tap image icon
3. Select any images
4. Collapse the selection pop up
5. Type any message into textbox -> tap 'send' button

**Actual result:**
The non-confirmed images are sent together with text
https://user-images.githubusercontent.com/52490791/208091697-8d42d4cb-185f-4374-9314-50c1f5352fd6.mp4


**Expected result:**
The images are not sent in the message if current images are not selected

**ENV:** All devices

**ISSUE 4: The selector is blinking after its opening if a lot of photos on the device are set**
**Steps to reproduce:**
1. Set at least 50 photos into your device
2. Open 1-1 chat
3. Open the selector

**Actual result:**
The selector is blinking after its opening
https://user-images.githubusercontent.com/52490791/208098745-95db2e9e-1e60-4d93-a386-e458cac2c664.mp4

**Expected result:**
The selector is not blinking

**ENV:** Iphone 11 pro v16

**ISSUE 5 the [send] button does not appear if only images are added** 
1. Open 1-1 chat
2. Open the selector
3. Select image
4. Confirm selection
5. Try to send the image without text

**Actual result:**
Send button is not shown if only images are selected
![image](https://user-images.githubusercontent.com/52490791/208102241-2459acc2-c295-4654-8f1b-d71e6ce91220.png)

**Expected result:**
The send button is shown if only images are in the textbox
![image](https://user-images.githubusercontent.com/52490791/208102730-6dbcf9d5-cb50-4fe1-81d0-93ac7c255b3a.png)

**ISSUE 6. Photo is not shown as selected if selector is reopened**
1. Open 1-1 chat
2. Open the selector
3. Select image
4. Confirm selection
5. Tap on the 'image' icon again to open the selector

**Actual result:**
The image is shown as unselected into the opened selector

https://user-images.githubusercontent.com/52490791/208105092-cf8e45b8-fe43-4b9a-8e0a-6f629e5120d8.mp4

**Expected result:**
The image is still shown as selected if current image is shown in the textbox and selector is reopened
![image](https://user-images.githubusercontent.com/52490791/208104915-9ae4aafe-29cd-4427-84db-0e1df661d153.png)

**ISSUE 7: All images are not shown into textbox after user select more than 5 photos**
**Steps to reproduce:**
1. Go to the 1-1 chat.
2.  Select at least 6-7 images
3. Confirm selection
4. Check count of selected images within textbox

**Actual result:**
only 5 photos are saved within textbox
https://user-images.githubusercontent.com/52490791/208107002-1f196e5e-1e9a-4951-a563-c759231336d1.mp4


**Expected result:**
All selected images are saved

**Questions:**
**1. Should the access error message be implemented in the scope of this PR?**
**Actual behavior:**
How it looks now if access is denied
**Expected behavior**
How design requires
![image](https://user-images.githubusercontent.com/52490791/208089747-b715910b-cc08-45ae-8ea5-44e3cf4a925e.png)

**2. Should the long press within selector on the image be implemented in the scope of this PR?**
**Actual result:**
Nothing happens after long press is done
**Expected behavior:**
![image](https://user-images.githubusercontent.com/52490791/208105677-730f8c02-40ba-468e-9be1-615a9841681c.png)

**3. Should the sent images be shown as per design in the scope of this PR? Currently, they are shown separately and the image is shown upper the image instead of under**
**Actual behavior:**
![image](https://user-images.githubusercontent.com/52490791/208109556-5d6ca815-fd43-4089-832a-0523bfa5b4ae.png)

**Expected behavior**
![image](https://user-images.githubusercontent.com/52490791/208109733-9cbc9228-e9ff-48e7-b889-240b0c354e0c.png)

status-im-auto commented:
## 0% of end-end tests have passed
```
Total executed tests: 2
Failed tests: 2
Passed tests: 0
```
```
IDs of failed tests: 702844,702840 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14487.apk&PR_ID=14487&TR_CASE_IDS=702844,702840">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068213">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Wait for element `Button` for max 30s and click when it is available </div></blockquote></p>

```
critical/test_public_chat_browsing.py:739: in test_community_links_with_previews_github_youtube_twitter_gif_send_enable
    self.channel_1.element_by_translation_id("enable-all").wait_and_click()
../views/base_element.py:370: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: Button by xpath:`//*[@text="Enable all"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/088870ee87c34f339e95ac8a8af978bd?auth=bb57efec6ab2ebb54a140a7e119c454f#160">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/088870ee87c34f339e95ac8a8af978bd/0056screenshot.png?auth=bb57efec6ab2ebb54a140a7e119c454f">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c82cdd59eb443c2850c26cf7fa04175?auth=08a4b7629401f8110ad2392d27c9d42c#77">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/7c82cdd59eb443c2850c26cf7fa04175/0026screenshot.png?auth=08a4b7629401f8110ad2392d27c9d42c">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068212">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `RepliedMessageText` by `xpath`: `//*[starts-with(@text,'test message')]/preceding::android.widget.TextView[@content-desc='quoted-message']` </div><div>Device 1: `RepliedMessageText` is `🇹🇻` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:693: in test_community_emoji_send_copy_paste_reply
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Emoji message was not copied
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/088870ee87c34f339e95ac8a8af978bd?auth=bb57efec6ab2ebb54a140a7e119c454f#133">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/088870ee87c34f339e95ac8a8af978bd/0056screenshot.png?auth=bb57efec6ab2ebb54a140a7e119c454f">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c82cdd59eb443c2850c26cf7fa04175?auth=08a4b7629401f8110ad2392d27c9d42c#64">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/7c82cdd59eb443c2850c26cf7fa04175/0026screenshot.png?auth=08a4b7629401f8110ad2392d27c9d42c">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details>

ibrkhalil commented on src/status_im/chat/models/images.cljs:
```diff
@@ -117,15 +120,32 @@
-  {:db (assoc db :camera-roll-photos (mapv #(get-in % [:node :image :uri]) photos))})
+  [{:keys [db] :as cofx} photos page-info end-cursor]
+  (let [photos_x (when end-cursor (:camera-roll/photos db))]
+    {:db (-> db
+             (assoc :camera-roll/photos (concat photos_x (map #(get-in % [:node :image :uri]) photos)))
+             (assoc :camera-roll/end-cursor (:end_cursor page-info))
+             (assoc :camera-roll/has-next-page (:has_next_page page-info))
+             (assoc :camera-roll/loading-more false))}))
```
Can this be done in one `assoc`?

ibrkhalil commented on src/status_im/ui2/screens/chat/composer/images/view.cljs:
```diff
@@ -0,0 +1,24 @@
+            [re-frame.core :as rf]))
+
+(defn image [item]
+  [rn/view
```
Can this be a React fragment ` [:<> ...]`?

OmarBasem committed:
rebase

OmarBasem committed:
android bottom button

OmarBasem committed:
qa fixes

OmarBasem commented:
@VladimrLitvinenko thanks for your reviews! I have fixed the issues. Please let me know if there are any more issues.

Issue 1: fixed (please note the images are not shown in the input on Android, which is not part of this PR)
Issue 2: a new bottom sheet is currently under development, not part of this PR
Issue 3: fixed
Issue 4: fixed
Issue 5: fixed
Issue 6: fixed
Issue 7: currently sending more than 5 photos is not supported, needs backend implementation
Questions 1,2,3: not part of this PR

status-im-auto commented:
## 84% of end-end tests have passed
```
Total executed tests: 19
Failed tests: 3
Passed tests: 16
Not executed tests: 1
```
```
IDs of not executed tests: 702847 
```
```
IDs of failed tests: 702840,702844,702745 
```
<h3>Not executed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14487.apk&PR_ID=14487&TR_CASE_IDS=702847">Rerun not executed tests</a></li></details><h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14487.apk&PR_ID=14487&TR_CASE_IDS=702840,702844,702745">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068475">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `RepliedMessageText` by `xpath`: `//*[starts-with(@text,'test message')]/preceding::android.widget.TextView[@content-desc='quoted-message']` </div><div>Device 1: `RepliedMessageText` is `🐉` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:693: in test_community_emoji_send_copy_paste_reply
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Emoji message was not copied
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#182">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29/0073screenshot.png?auth=4d4dbb7fd071416e30f0479e520266b0">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#112">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03/0051screenshot.png?auth=bf82bbf89c051a43319876802a29a25e">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068479">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><p><blockquote><div>Device 2: Getting preview message for link: https://giphy.com/gifs/this-is-fine-QMHoU66sBXqqLqYvGO </div><div>Device 2: Scrolling down to PreviewImage </div></blockquote></p>

```
critical/test_public_chat_browsing.py:758: in test_community_links_with_previews_github_youtube_twitter_gif_send_enable
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Enable button is still shown after clicking on 'Don't ask again'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#239">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29/0073screenshot.png?auth=4d4dbb7fd071416e30f0479e520266b0">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#133">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03/0051screenshot.png?auth=bf82bbf89c051a43319876802a29a25e">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068467">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `MemberPhoto` by `xpath`: `//*[starts-with(@text,'profile_photo')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='user-avatar']` </div><div>Device 2: Image differs from template to 20.02664005055147 percents </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1414: in test_1_1_chat_non_latin_messages_stack_update_profile_photo
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Image of user in 1-1 chat is too different from template!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/63775e1726604efbba4ba1a7fa734ef0?auth=872c0e28db6b0a05a26fd293198d815c#192">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/63775e1726604efbba4ba1a7fa734ef0/0070screenshot.png?auth=872c0e28db6b0a05a26fd293198d815c">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/830d1827c78641b496bcc892d828f0ea?auth=d1a8c12e1d94e27da58fc452adf16779#109">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/830d1827c78641b496bcc892d828f0ea/0049screenshot.png?auth=d1a8c12e1d94e27da58fc452adf16779">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (16)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068466">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/63775e1726604efbba4ba1a7fa734ef0?auth=872c0e28db6b0a05a26fd293198d815c#271">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/830d1827c78641b496bcc892d828f0ea?auth=d1a8c12e1d94e27da58fc452adf16779#174">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068463">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/63775e1726604efbba4ba1a7fa734ef0?auth=872c0e28db6b0a05a26fd293198d815c#84">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/830d1827c78641b496bcc892d828f0ea?auth=d1a8c12e1d94e27da58fc452adf16779#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068464">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/63775e1726604efbba4ba1a7fa734ef0?auth=872c0e28db6b0a05a26fd293198d815c#104">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/830d1827c78641b496bcc892d828f0ea?auth=d1a8c12e1d94e27da58fc452adf16779#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068471">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1493bc9fac6840dc9c7d4da734e38cbe?auth=89a9301e02a2e9b42ee8fd37abe7c833#111">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/759d538093b640c4b6ee6a0af86dc01f?auth=60858b06e5a158b20e75b573cc3f521a#63">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a136c608b7c14e799d605218c8bbe2fd?auth=9dbb9592883eeb43ee94d056f3ed8f23#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068472">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1493bc9fac6840dc9c7d4da734e38cbe?auth=89a9301e02a2e9b42ee8fd37abe7c833#128">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/759d538093b640c4b6ee6a0af86dc01f?auth=60858b06e5a158b20e75b573cc3f521a#79">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a136c608b7c14e799d605218c8bbe2fd?auth=9dbb9592883eeb43ee94d056f3ed8f23#68">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068470">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7977ff80312b423bbdcd299ef047f09b?auth=cdcaaa419b3d78578714cd318381c8f5#80">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068468">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7977ff80312b423bbdcd299ef047f09b?auth=cdcaaa419b3d78578714cd318381c8f5#41">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068469">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7977ff80312b423bbdcd299ef047f09b?auth=cdcaaa419b3d78578714cd318381c8f5#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068465">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/72932f66b831458db0052943b25d8294?auth=f869c997849f11324a289d7afa36d50a#65">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/612d7746fdb5475bb46b5f360c166a96?auth=7ca5e747cdb8f6b8f950d2185a91ee47#60">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068481">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a75904e11e9e44e093fdcdb190009677?auth=91f667c210de6f5fd33de70a230700ab#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068480">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#286">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#193">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068476">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#209">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068474">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#161">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068477">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#266">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#185">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068473">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068478">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b8652e3062f34d1282f6e920468cdf29?auth=4d4dbb7fd071416e30f0479e520266b0#146">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/79cd5e39ff4f4a2a945f6d341be1da03?auth=bf82bbf89c051a43319876802a29a25e#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

VladimrLitvinenko commented:
@OmarBasem thank you for PR. Issues 3,4,5,6 are fixed and PR ready to be merged
Take a look, one small priority issue still exist. I believe it can be fixed separately as follow up https://github.com/status-im/status-mobile/issues/14580

OmarBasem commented:
Thank you @VladimrLitvinenko for your reviews!

OmarBasem commented:
@flexsurfer I need to merge this one now for a dependency PR. If there is still any issue with this PR please let me know and I will fix it in the following PR.
